### PR TITLE
TRUS-3685 Add TaxLiabilityYesNo and BeforeYouContinueToTaxable page

### PR DESCRIPTION
**Pull request by carrielies**

### CODE REVIEW

File: app/controllers/transition/TaxLiabilityYesNoController.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class TaxLiabilityYesNoController @Inject()(
+                                       override val messagesApi: MessagesApi,
+                                       playbackRepository: PlaybackRepository,
+                                       actions: Actions,
+                                       val controllerComponents: MessagesControllerComponents,
+                                       yesNoFormProvider: YesNoFormProvider,
+                                       view: TaxLiabilityYesNoView,
```
Comment (jackheslop96) [inline app/controllers/transition/TaxLiabilityYesNoController.scala:40]: Comma causing build to fail

File: app/controllers/transition/TaxLiabilityYesNoController.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      form.bindFromRequest().fold(
+        (formWithErrors: Form[_]) => {
+          val identifier = request.userAnswers.identifier
+          val identifierType = request.userAnswers.identifierType
+          Future.successful(BadRequest(view(formWithErrors, identifier, identifierType)))
+        },
+        value => {
+          if (value) {
+            for {
+              updatedAnswers <- Future.fromTry(request.userAnswers.set(TaxLiabilityYesNoPage, value))
```
Comment (jackheslop96) [inline app/controllers/transition/TaxLiabilityYesNoController.scala:72]: Should we be setting this when No answered as well? Just thinking if they select the 'Needs to pay tax' again they'd probably want to see this question pre-populated

File: app/controllers/transition/TaxLiabilityYesNoController.scala (hunk 1 above)
Comment (carrielies) [reply app/controllers/transition/TaxLiabilityYesNoController.scala:40]: Last minute tidy up causing the problem

File: app/controllers/transition/TaxLiabilityYesNoController.scala (hunk 2 above)
Comment (carrielies) [reply app/controllers/transition/TaxLiabilityYesNoController.scala]: Good point. Done

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**